compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -569,24 +562,21 @@ convertModule lfVersion pkgMap stablePackages isGenerated file x depOrphanModule
         emptyInterfaces :: S.Set (GHC.Module, TypeConName)
         emptyInterfaces = S.fromList
           [ (mod, mkTypeCon [getOccText ifc])
-          | (name, val) <- binds
-          -- We're looking for `instance DA.Internal.Desugar.Implements I I`
-          , DFunId _ <- [idDetails name]
-          , TypeCon implementsCls [TypeCon tpl [], TypeCon ifc []] <- [varType name]
-          , NameIn DA_Internal_Desugar "Implements" <- [implementsCls]
-          , Just mod <- [nameModule_maybe (getName ifc)]
-          , tpl == ifc
+          |
+          -- We're looking for `data DamlInterface => I = ...`
+            (_, ifc) <- MS.toList interfaceCons
           -- such that there are no `instance HasMethod I _ _`
           , null
             [ ()
-            | (name, val) <- binds
+            | (name, _) <- binds
             , DFunId _ <- [idDetails name]
             , TypeCon hasMethodCls
```

Comment:
Is hasMethodCls constrained by anything here? This seems a bit strange.

Reply:
thanks for spotting this! I'm not sure how I missed it 🤔

Reply:
done now!